plugins/search-react/src/components/SearchBar/SearchBar.tsx

```diff
@@ -64,94 +64,96 @@ export type SearchBarBaseProps = Omit<InputBaseProps, 'onChange'> & {
+      );
+    }, [defaultValue]);
+
+    useDebounce(() => onChange(value), debounceTime, [value]);
+
+    const handleChange = useCallback(
+      (e: ChangeEvent<HTMLInputElement>) => {
+        setValue(e.target.value);
+      },
+      [setValue],
     );
-  }, [defaultValue]);
-
-  useDebounce(() => onChange(value), debounceTime, [value]);
-
-  const handleChange = useCallback(
-    (e: ChangeEvent<HTMLInputElement>) => {
-      setValue(e.target.value);
-    },
-    [setValue],
-  );
-
-  const handleKeyDown = useCallback(
-    (e: KeyboardEvent<HTMLInputElement>) => {
-      if (onKeyDown) onKeyDown(e);
-      if (onSubmit && e.key === 'Enter') {
-        onSubmit();
+
+    const handleKeyDown = useCallback(
+      (e: KeyboardEvent<HTMLInputElement>) => {
+        if (onKeyDown) onKeyDown(e);
+        if (onSubmit && e.key === 'Enter') {
+          onSubmit();
+        }
+      },
+      [onKeyDown, onSubmit],
+    );
+
+    const handleClear = useCallback(() => {
+      onChange('');
+      if (onClear) {
+        onClear();
       }
-    },
-    [onKeyDown, onSubmit],
-  );
-
-  const handleClear = useCallback(() => {
-    onChange('');
-  }, [onChange]);
-
-  const placeholder = `Search in ${
-    configApi.getOptionalString('app.title') || 'Backstage'
-  }`;
-
-  const startAdornment = (
-    <InputAdornment position="start">
-      <IconButton aria-label="Query" disabled>
-        <SearchIcon />
-      </IconButton>
-    </InputAdornment>
-  );
-
-  const endAdornment = (
-    <InputAdornment position="end">
-      <IconButton aria-label="Clear" onClick={handleClear}>
-        <ClearButton />
-      </IconButton>
-    </InputAdornment>
-  );
-
-  const searchBar = (
-    <TrackSearch>
-      <InputBase
-        data-testid="search-bar-next"
-        value={value}
-        placeholder={placeholder}
-        startAdornment={startAdornment}
-        endAdornment={clearButton ? endAdornment : defaultEndAdornment}
-        inputProps={{ 'aria-label': 'Search', ...defaultInputProps }}
-        fullWidth={fullWidth}
-        onChange={handleChange}
-        onKeyDown={handleKeyDown}
-        {...props}
-      />
-    </TrackSearch>
-  );
-
-  return hasSearchContext ? (
-    searchBar
-  ) : (
-    <SearchContextProvider>{searchBar}</SearchContextProvider>
-  );
-};
+    }, [onChange, onClear]);
+
+    const placeholder =
+      defaultPlaceholder ??
+      `Search in ${configApi.getOptionalString('app.title') || 'Backstage'}`;
+
+    const startAdornment = (
+      <InputAdornment position="start">
+        <IconButton aria-label="Query" size="small" disabled>
+          <SearchIcon />
+        </IconButton>
+      </InputAdornment>
+    );
+
+    const endAdornment = (
+      <InputAdornment position="end">
+        <IconButton aria-label="Clear" size="small" onClick={handleClear}>
+          <ClearButton />
+        </IconButton>
+      </InputAdornment>
+    );
+
+    return (
```

Comment:
this was wrapped with the search context before, but I don't see that anymore, was that changed on purpose? If so, do we need to clarify that in a changeset as the SearchBarBase is exported publicly? just want to confirm.

Reply:
As the search input base is recommended for when we are not using the search context I had removed the context from it, but it seems that TrackSearch consumes the term from the context... It would be better to change here and add context in SearchBar or change TrackSearch to also accept the term via prop?

Reply:
For now, I've added the context back, and we can rethink the search track in the future if necessary

Reply:
Sounds good! We can do that in a separate PR to keep this one focused! 💯